hw_lesson17.py

```diff
@@ -0,0 +1,151 @@
+import re
+from functools import wraps
+
+"""
+Task 1
+Create a class method named `validate`, which should be called from the `__init__` method to validate parameter email,
+passed to the constructor.
+The logic inside the `validate` method could be to check if the passed email parameter is a valid email string.
+"""
+
+
+class Validate:
+
+    def __init__(self, validate_email: str):
+        if not isinstance(validate_email, str):
+            raise TypeError('Error type email!!!')
+
+        if not re.match('\b[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\.[A-Z|a-z]{2,7}\b', validate_email):
+            raise TypeError('Error type email!!!')
+
+        self.validate_email = validate_email
+
+
+"""
+Task 2
+Implement 2 classes, the first one is the Boss and the second one is the Worker.
+
+Worker has a property 'boss', and its value must be an instance of Boss.
+
+You can reassign this value, but you should check whether the new value is Boss.
+Each Boss has a list of his own workers. You should implement a method that allows you to add workers to a Boss.
+You're not allowed to add instances of Boss class to workers list directly via access to attribute,
+use getters and setters instead!
+"""
+
+
+class Boss:
+    def __init__(self, id_: int, name: str, company: str):
+        self.id = id_
+        self.name = name
+        self.company = company
+        self.workers = []
+
+    @property
```

Comment:
Now Boss has property boss. So, to add a worker for a boss I have to make something like that, is it obvious?
```
worker = Worker()
boss = Boss()
boss.boss = worker  # Looks strange, doesn't it?
```